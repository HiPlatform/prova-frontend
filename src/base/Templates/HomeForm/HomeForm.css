.homeForm {
  display: block;
  background-color: #f5f5f5;
  border-radius: 5px;
  margin: 40px 16px 40px 0px;
  width: 90%;
  box-shadow: 0px 3px 3px rgba(0, 0, 0, 0.1);
  position: relative;
  padding: 10px 15px;
  float: left; }
  .homeForm > input {
    margin: 10px 0px; }
  @media screen and (min-width: 341px) {
    .homeForm {
      display: flex;
      flex-direction: row; }
      .homeForm > .buttonGroup {
        flex: 3; }
      .homeForm > input {
        flex: 3;
        margin: 0px 15px; }
      .homeForm > .uiButton {
        flex: 1; } }

.logo {
  text-align: center; }
