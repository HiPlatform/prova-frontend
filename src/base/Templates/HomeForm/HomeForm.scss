.homeForm {
  display: block;
  background-color: #f5f5f5;
  border-radius: 5px;
  margin: 40px 16px 40px 0px;
  width: 90%;
  box-shadow: 0px 3px 3px rgba(0,0,0,0.1);
  position: relative;
  padding: 10px 15px;
  float: left;

  & > input {
    margin: 10px 0px;
  }

  @media screen and (min-width: 768px) {
    display: flex;
    flex-direction: row;

    & > .buttonGroup {
      flex: 3;
    }

    & > input {
      flex: 3;
      margin: 0px 15px;
    }

    & > .uiButton {
      flex: 1;
    }
  }
}

.logo {
  text-align: center;
}