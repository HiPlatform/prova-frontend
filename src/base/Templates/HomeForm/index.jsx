import React from 'react';
import PropTypes from 'prop-types';

import Button from '../../Atoms/Button';
import Input from '../../Atoms/Input';
import ButtonGroup from '../../Molecules/ButtonGroup';

import './HomeForm.css';

const HomeForm = ({
  value,
  searchType,
  fnType,
  fnInput,
  fnSearch,
}) => (
  <div>
    <h1 className="logo">Music Tracker</h1>
    <div className="homeForm">
      <ButtonGroup types={['artist', 'album', 'track']} change={fnType} active={searchType} />
      <Input type="text" value={value} onChange={fnInput} />
      <Button onClick={() => fnSearch(searchType, value)} modifier="solid">SEARCH</Button>
    </div>
  </div>
);

HomeForm.defaultProps = {
  value: '',
  searchType: '',
};

HomeForm.propTypes = {
  value: PropTypes.string,
  searchType: PropTypes.string,
  fnType: PropTypes.func.isRequired,
  fnInput: PropTypes.func.isRequired,
  fnSearch: PropTypes.func.isRequired,
};

export default HomeForm;
