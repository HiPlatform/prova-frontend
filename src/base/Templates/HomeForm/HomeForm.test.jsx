import React from 'react';
import { shallow } from 'enzyme';
import HomeForm from './index';

describe('HomeForm', () => {
  const fnType = jest.fn();
  const fnInput = jest.fn();
  const fnSearch = jest.fn();
  const props = {
    fnType,
    fnInput,
    fnSearch,
  };
  const component = shallow(<HomeForm {...props} />);

  it('should render without crashing', () => {
    expect(component).toMatchSnapshot();
  });

  it('should call the search function when the user clicks on the button', () => {
    component.find('Button').simulate('click');
    expect(fnSearch).toHaveBeenCalled();
  });
});
