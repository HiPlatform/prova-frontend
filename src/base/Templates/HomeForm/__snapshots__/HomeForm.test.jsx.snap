// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeForm should render without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HomeForm
    fnInput={[Function]}
    fnSearch={[Function]}
    fnType={[Function]}
    searchType=""
    value=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="logo"
>
          Music Tracker
</h1>,
        <div
          className="homeForm"
>
          <ButtonGroup
                    active=""
                    change={[Function]}
                    types={
                              Array [
                                        "artist",
                                        "album",
                                        "track",
                                      ]
                    }
          />
          <Input
                    onChange={[Function]}
                    type="text"
                    value=""
          />
          <Button
                    modifier="solid"
                    onClick={[Function]}
          >
                    SEARCH
          </Button>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Music Tracker",
          "className": "logo",
        },
        "ref": null,
        "rendered": "Music Tracker",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <ButtonGroup
              active=""
              change={[Function]}
              types={
                            Array [
                                          "artist",
                                          "album",
                                          "track",
                                        ]
              }
/>,
            <Input
              onChange={[Function]}
              type="text"
              value=""
/>,
            <Button
              modifier="solid"
              onClick={[Function]}
>
              SEARCH
</Button>,
          ],
          "className": "homeForm",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "active": "",
              "change": [Function],
              "types": Array [
                "artist",
                "album",
                "track",
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "onChange": [Function],
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "SEARCH",
              "modifier": "solid",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "SEARCH",
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="logo"
>
            Music Tracker
</h1>,
          <div
            className="homeForm"
>
            <ButtonGroup
                        active=""
                        change={[Function]}
                        types={
                                    Array [
                                                "artist",
                                                "album",
                                                "track",
                                              ]
                        }
            />
            <Input
                        onChange={[Function]}
                        type="text"
                        value=""
            />
            <Button
                        modifier="solid"
                        onClick={[Function]}
            >
                        SEARCH
            </Button>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Music Tracker",
            "className": "logo",
          },
          "ref": null,
          "rendered": "Music Tracker",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <ButtonGroup
                active=""
                change={[Function]}
                types={
                                Array [
                                                "artist",
                                                "album",
                                                "track",
                                              ]
                }
/>,
              <Input
                onChange={[Function]}
                type="text"
                value=""
/>,
              <Button
                modifier="solid"
                onClick={[Function]}
>
                SEARCH
</Button>,
            ],
            "className": "homeForm",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "active": "",
                "change": [Function],
                "types": Array [
                  "artist",
                  "album",
                  "track",
                ],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "onChange": [Function],
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "SEARCH",
                "modifier": "solid",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "SEARCH",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
